src/main/java/uk/gov/ons/ctp/common/state/BasicStateTransitionManager.java

```diff
@@ -38,7 +40,13 @@ public S transition(final S sourceState, final E event) throws CTPException {
       destinationState = outputMap.get(event);
     }
     if (destinationState == null) {
```

Comment:
In the event of a dodgy transition, destinationState is not null but the same state as it was previously so this check should check equality of new and old states.